extensions/pkg/controller/worker/genericactuator/actuator_delete.go

```diff
@@ -101,6 +102,9 @@ func (a *genericActuator) Delete(ctx context.Context, log logr.Logger, worker *e
 
 	// Wait until all machine resources have been properly deleted.
 	if err := a.waitUntilMachineResourcesDeleted(ctx, log, worker, workerDelegate); err != nil {
+		if a.errorCodeCheckFunc != nil {
+			return v1beta1helper.NewErrorWithCodes(fmt.Errorf("failed while waiting for all machine resources to be deleted: %w", err), a.errorCodeCheckFunc(err)...)
+		}
 		return fmt.Errorf("failed while waiting for all machine resources to be deleted: %w", err)
```

Comment:
Store `fmt.Errorf("failed while waiting for all machine resources to be deleted: %w", err)` in a variable and reuse it